internal/appsec/dyngo/instrumentation/httpsec/actions.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package httpsec
+
+import (
+	"net/http"
+)
+
+// Action is used to identify any action kind
+type Action interface {
+	isAction()
+}
+
+// BlockRequestAction is the parameter struct used to perform actions of kind ActionBlockRequest
+type BlockRequestAction struct {
+	Action
```

Comment:
```suggestion
	Action // to implement the interface
```

Reply:
BTW, I was thinking of what are the pros and cons of this "hack" and not that you are impacting the sizeof of your struct (IIRC, an interface is made of 2 pointers), while implementing the method doesn't impact the size of the struct ;-)

Reply:
Let's stick to implementing only then :+1: 
```suggestion
```